sys/net/link_layer/ieee802154/security.c

```diff
@@ -24,16 +24,51 @@
 #include "net/ieee802154_security.h"
 
 const ieee802154_radio_cipher_ops_t ieee802154_radio_cipher_ops = {
-    .set_key = ieee802154_sec_set_key,
-    .ecb = ieee802154_sec_ecb,
-    .cbc = ieee802154_sec_cbc
+    .set_key = NULL,
+    .ecb = NULL,
+    .cbc = NULL
```

Comment:
Why not

```suggestion
    .set_key = _sec_key,
    .ecb = _sec_ebc,
    .cbc = _sec_cbc
```
The functions are defined in the same file. Just move the `ieee802154_radio_cipher_ops` definition below their declarations.

Reply:
First, because of clarity. It think it is clearer to see in `security.c` that the fallback is used when no override is defined.
Also, your proposal seems to be increasing ROM usage again by 4 bytes, compare to the PR.
Finally the implementation of `security.c` can be potentially further optimized if we keep the `if`s, by using compile-time flags that indicate that no hardware acceleration is present on the current hardware configuration. In that case the compiler has even more information and would remove the pointer call entirely.

Reply:
> your proposal seems to be increasing ROM usage again by 4 bytes, compare to the PR.

Huh, that's surprising - but then again some of these functions are no-ops, so the compiler can now remove them altogether.